src/ray/gcs/store_client/in_memory_store_client.cc

```diff
@@ -37,37 +40,45 @@ Status InMemoryStoreClient::AsyncPutWithIndex(const std::string &table_name,
   absl::MutexLock lock(&(table->mutex_));
   table->records_[key] = data;
   table->index_keys_[index_key].emplace_back(key);
-  main_io_service_.post([callback]() { callback(Status::OK()); },
-                        "GcsInMemoryStore.PutWithIndex");
+  if (callback != nullptr) {
+    main_io_service_.post([callback]() { callback(Status::OK()); },
+                          "GcsInMemoryStore.PutWithIndex");
+  }
   return Status::OK();
 }
 
 Status InMemoryStoreClient::AsyncGet(const std::string &table_name,
                                      const std::string &key,
                                      const OptionalItemCallback<std::string> &callback) {
+  RAY_CHECK(callback != nullptr);
   auto table = GetOrCreateTable(table_name);
   absl::MutexLock lock(&(table->mutex_));
   auto iter = table->records_.find(key);
+  boost::optional<std::string> data;
   if (iter != table->records_.end()) {
-    auto data = iter->second;
-    main_io_service_.post([callback, data]() { callback(Status::OK(), data); },
-                          "GcsInMemoryStore.Get");
-  } else {
-    main_io_service_.post([callback]() { callback(Status::OK(), boost::none); },
-                          "GcsInMemoryStore.GetNone");
+    data = iter->second;
   }
+
+  main_io_service_.post(
+      [callback, data = std::move(data)]() { callback(Status::OK(), data); },
+      "GcsInMemoryStore.Get");
+
   return Status::OK();
 }
 
 Status InMemoryStoreClient::AsyncGetAll(
     const std::string &table_name,
     const MapCallback<std::string, std::string> &callback) {
+  RAY_CHECK(callback);
   auto table = GetOrCreateTable(table_name);
   absl::MutexLock lock(&(table->mutex_));
-  auto result = std::make_shared<std::unordered_map<std::string, std::string>>();
-  result->insert(table->records_.begin(), table->records_.end());
-  main_io_service_.post([result, callback]() { callback(std::move(*result)); },
-                        "GcsInMemoryStore.GetAll");
+  auto result = std::unordered_map<std::string, std::string>();
+  result.insert(table->records_.begin(), table->records_.end());
+  if (callback != nullptr) {
```

Comment:
`callback` cannot be `nullptr` in this function? Or remove `RAY_CHECK(callback);`?